tests/dependency_injection/test_request_local_caching.py

```diff
@@ -0,0 +1,34 @@
+from starlite import Provide, create_test_client, get
+from starlite.status_codes import HTTP_200_OK
+
+
+def test_caching_per_request() -> None:
```

Comment:
I thought dependenc caching is gone 😉 , in fact i know its gone. 

I think you should probably rebase and remove this.

What I am not seeing-- tests that ensure the functionality you are adding works as expected. And I would like to run benchmarks to ensure that this improves performance and doesn't cause memory issues.

Reply:
Dependency "caching" gets reintroduced (every dependency is only resolved once), so I reused the test. Maybe change the name to be more clear?

I added a test for the batching.